packages/dx-react-grid-demos/src/demo-sources/grid-lazy-loading/remote-data.jsxt

```diff
@@ -42,7 +44,70 @@ const DateTypeProvider = props => (
   />
 );
 
+function reducer(state, { type, payload }) {
+  switch (type) {
+    case 'UPDATE_ROWS':
+      return {
+        ...state,
+        ...payload,
+        loading: false,
+      };
+    case 'CHANGE_SORTING':
+      return {
+        ...state,
+        forceReload: true,
+        rows: [],
+        sorting: payload,
+      };
+    case 'CHANGE_FILTERS':
+      return {
+        ...state,
+        forceReload: true,
+        requestedSkip: 0,
+        rows: [],
+        filters: payload,
+      };
+    case 'START_LOADING':
+      return {
+        ...state,
+        requestedSkip: payload.requestedSkip,
+        take: payload.take,
+      };
+    case 'REQUEST_ERROR':
+      return {
+        ...state,
+        loading: false,
+      };
+    case 'FETCH_INIT':
+      return {
+        ...state,
+        loading: true,
+        forceReload: false,
+      };
+    case 'UPDATE_QUERY':
+      return {
+        ...state,
+        lastQuery: payload,
+      };
+    default:
+      return state;
+  }
+}
+
 export default () => {
+  const initialState = {
```

Comment:
Should `InitialState` be inside the render method?

Reply:
We can move it outside